### PR TITLE
Increase test coverage for Product entity in API

### LINKED ISSUE #3303: Increase coverage for Product entity

### PR DESCRIPTION
**Pull request by oshtaier**
Closes #3303
```
nosetests tests/foreman/api/test_product.py
..............S
----------------------------------------------------------------------
Ran 15 tests in 353.803s

OK (SKIP=1)
```

### CODE REVIEW

**Comment by sthirugn** [comment]
ACK

File: tests/foreman/api/test_product.py [hunk 1]
```diff
@@ -64,46 +80,91 @@ def test_positive_create_with_gpg(self):
...
-                self.assertEqual(
-                    getattr(self.product, attrs.keys()[0]),
-                    attrs.values()[0]
-                )
+        product = entities.Product(organization=self.org).create()
+        for new_name in valid_data_list():
+            with self.subTest(new_name):
+                entities.Product(id=product.id, name=new_name).update(['name'])
+                updated = entities.Product(id=product.id).read()
+                self.assertEqual(new_name, updated.name)
```
Comment (abalakh) [inline tests/foreman/api/test_product.py]: No need to create 2 new entities (and single-use variables) when you already have the required one:
```python
        product = entities.Product(organization=self.org).create()
        for new_name in valid_data_list():
            with self.subTest(new_name):
                product.name = new_name
                product = product.update(['name'])
                self.assertEqual(new_name, product.name)
```
Comment applies to other tests as well.

File: tests/foreman/api/test_product.py [hunk 2]
```diff
@@ -64,46 +80,91 @@ def test_positive_create_with_gpg(self):
...
+        """
+        product = entities.Product(organization=self.org).create()
+        for new_desc in valid_data_list():
+            with self.subTest(new_desc):
+                entities.Product(
+                    id=product.id,
+                    description=new_desc,
+                ).update(['description'])
+                updated = entities.Product(id=product.id).read()
+                self.assertEqual(new_desc, updated.description)
```
Comment (abalakh) [inline tests/foreman/api/test_product.py]: Same here

File: tests/foreman/api/test_product.py [hunk 3]
```diff
@@ -111,18 +172,109 @@ def test_positive_update_name_to_original(self):
...
+        """Attempt to update product label to another one.
+
+        @Assert: Product is not updated and error is raised
+
+        @Feature: Product
+        """
+        product = entities.Product(organization=self.org).create()
+        with self.assertRaises(HTTPError):
+            entities.Product(
+                id=product.id, label=gen_string('alpha')).update(['label'])
```
Comment (abalakh) [inline tests/foreman/api/test_product.py]: And here

**Comment by abalakh** [comment]
ACK pending comments

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/api/test_product.py -m test_positive_update_name
..
----------------------------------------------------------------------
Ran 2 tests in 43.868s

OK
```
```
nosetests tests/foreman/api/test_product.py -m test_positive_update_description
.
----------------------------------------------------------------------
Ran 1 test in 53.296s

OK
```
```
nosetests tests/foreman/api/test_product.py -m test_negative_update_label
.
----------------------------------------------------------------------
Ran 1 test in 11.565s

OK
```

**Comment by oshtaier** [comment]
Addressed all comments

**Pull request merged**